src/type.js

```diff
@@ -297,6 +310,8 @@ function typeCharacter(
         newEntry = `-${char}`
       } else if (prevWasPeriod) {
         newEntry = `${prevValue}.${char}`
+      } else if (isValidDateValue(currentElement(), textToBeTyped)) {
```

Comment:
Should this be done whenever we have a valide date value? If so, we should put it in a plain `if` after the prevWas checks.